services/director-v2/src/simcore_service_director_v2/modules/redis.py

```diff
@@ -0,0 +1,180 @@
+import asyncio
+import logging
+from asyncio import Task
+from contextlib import asynccontextmanager, suppress
+from dataclasses import dataclass
+from functools import cached_property
+from typing import AsyncIterator, Final, Optional
+
+from fastapi import FastAPI
+from pydantic import NonNegativeInt
+from redis.asyncio import Redis
+from redis.asyncio.lock import Lock
+from settings_library.redis import RedisSettings
+from tenacity import retry
+from tenacity.before_sleep import before_sleep_log
+from tenacity.wait import wait_random
+
+from ..core.errors import ConfigurationError
+
+DEFAULT_LOCKS_PER_NODE: Final[int] = 2
+EXTEND_TASK_ATTR_NAME = "extend_task"
+
+DockerNodeId = str
+
+
+logger = logging.getLogger(__name__)
+
+redis_retry_policy = dict(
+    wait=wait_random(5, 10),
+    before_sleep=before_sleep_log(logger, logging.WARNING),
+    reraise=True,
+)
+
+
+def setup(app: FastAPI):
+    @retry(**redis_retry_policy)
+    async def on_startup() -> None:
+        app.state.redis_lock_manager = await RedisLockManager.create(app)
+
+    async def on_shutdown() -> None:
+        lock_manager: RedisLockManager = app.state.redis_lock_manager
+        await lock_manager.close()
+
+    app.add_event_handler("startup", on_startup)
+    app.add_event_handler("shutdown", on_shutdown)
+
+
+@dataclass
+class LocksPerNodeProvider:
+    # NOTE: added here for future use, in this iteration just
+    # returns a constant
+
+    async def get(  # pylint: disable=unused-argument,no-self-use
+        self, docker_node_id: DockerNodeId
+    ) -> int:
+        return DEFAULT_LOCKS_PER_NODE
```

Comment:
is dataclass idea not to replace getters?

Reply:
I was not that convinced at all myself. let's go fo dropping the complexity and adding a a note on how to implement it in the future. Since it will most likely change a lot.